// @speckle/shared/dist/commonjs/authz/domain/loaders.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Loaders = exports.AuthCheckContextLoaderKeys = void 0;
/**
 * All loaders must be listed here for app startup validation to work properly
 */
/* v8 ignore start  */
exports.AuthCheckContextLoaderKeys = {
    getEnv: 'getEnv',
    getProject: 'getProject',
    getProjectRoleCounts: 'getProjectRoleCounts',
    getProjectRole: 'getProjectRole',
    getProjectModelCount: 'getProjectModelCount',
    getServerRole: 'getServerRole',
    getWorkspace: 'getWorkspace',
    getWorkspaceRole: 'getWorkspaceRole',
    getWorkspaceSeat: 'getWorkspaceSeat',
    getWorkspaceModelCount: 'getWorkspaceModelCount',
    getWorkspaceProjectCount: 'getWorkspaceProjectCount',
    getWorkspacePlan: 'getWorkspacePlan',
    getWorkspaceLimits: 'getWorkspaceLimits',
    getWorkspaceSsoProvider: 'getWorkspaceSsoProvider',
    getWorkspaceSsoSession: 'getWorkspaceSsoSession',
    getAdminOverrideEnabled: 'getAdminOverrideEnabled',
    getComment: 'getComment',
    getModel: 'getModel',
    getVersion: 'getVersion'
};
exports.Loaders = exports.AuthCheckContextLoaderKeys; // shorter alias
