// @speckle/shared/dist/commonjs/authz/policies/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.authPoliciesFactory = void 0;
const canCreateWorkspaceProject_js_1 = require("./workspace/canCreateWorkspaceProject.js");
const canRead_js_1 = require("./project/canRead.js");
const canCreate_js_1 = require("./project/model/canCreate.js");
const canMoveToWorkspace_js_1 = require("./project/canMoveToWorkspace.js");
const canCreatePersonal_js_1 = require("./project/canCreatePersonal.js");
const canUpdate_js_1 = require("./project/canUpdate.js");
const canReadSettings_js_1 = require("./project/canReadSettings.js");
const canReadWebhooks_js_1 = require("./project/canReadWebhooks.js");
const canUpdateAllowPublicComments_js_1 = require("./project/canUpdateAllowPublicComments.js");
const canLeave_js_1 = require("./project/canLeave.js");
const canInvite_js_1 = require("./workspace/canInvite.js");
const canInvite_js_2 = require("./project/canInvite.js");
const canBroadcastActivity_js_1 = require("./project/canBroadcastActivity.js");
const canCreate_js_2 = require("./project/comment/canCreate.js");
const canArchive_js_1 = require("./project/comment/canArchive.js");
const canEdit_js_1 = require("./project/comment/canEdit.js");
const canUpdate_js_2 = require("./project/model/canUpdate.js");
const canDelete_js_1 = require("./project/model/canDelete.js");
const canCreate_js_3 = require("./project/version/canCreate.js");
const canUpdate_js_3 = require("./project/version/canUpdate.js");
const canReceive_js_1 = require("./project/version/canReceive.js");
const canRequestRender_js_1 = require("./project/version/canRequestRender.js");
const canCreate_js_4 = require("./project/automation/canCreate.js");
const canUpdate_js_4 = require("./project/automation/canUpdate.js");
const canRead_js_2 = require("./project/automation/canRead.js");
const canReceiveProjectsUpdatedMessage_js_1 = require("./workspace/canReceiveProjectsUpdatedMessage.js");
const canDelete_js_2 = require("./project/canDelete.js");
const canDelete_js_3 = require("./project/automation/canDelete.js");
const canPublish_js_1 = require("./project/canPublish.js");
const canLoad_js_1 = require("./project/canLoad.js");
const canUpdateEmbedOptions_js_1 = require("./workspace/canUpdateEmbedOptions.js");
const authPoliciesFactory = (loaders) => ({
    project: {
        automation: {
            canCreate: (0, canCreate_js_4.canCreateAutomationPolicy)(loaders),
            canRead: (0, canRead_js_2.canReadAutomationPolicy)(loaders),
            canUpdate: (0, canUpdate_js_4.canUpdateAutomationPolicy)(loaders),
            canDelete: (0, canDelete_js_3.canDeleteAutomationPolicy)(loaders)
        },
        model: {
            canCreate: (0, canCreate_js_1.canCreateModelPolicy)(loaders),
            canUpdate: (0, canUpdate_js_2.canUpdateModelPolicy)(loaders),
            canDelete: (0, canDelete_js_1.canDeleteModelPolicy)(loaders)
        },
        comment: {
            canCreate: (0, canCreate_js_2.canCreateProjectCommentPolicy)(loaders),
            canArchive: (0, canArchive_js_1.canArchiveProjectCommentPolicy)(loaders),
            canEdit: (0, canEdit_js_1.canEditProjectCommentPolicy)(loaders)
        },
        version: {
            canCreate: (0, canCreate_js_3.canCreateProjectVersionPolicy)(loaders),
            canUpdate: (0, canUpdate_js_3.canUpdateProjectVersionPolicy)(loaders),
            canReceive: (0, canReceive_js_1.canReceiveProjectVersionPolicy)(loaders),
            canRequestRender: (0, canRequestRender_js_1.canRequestProjectVersionRenderPolicy)(loaders)
        },
        canBroadcastActivity: (0, canBroadcastActivity_js_1.canBroadcastProjectActivityPolicy)(loaders),
        canRead: (0, canRead_js_1.canReadProjectPolicy)(loaders),
        canMoveToWorkspace: (0, canMoveToWorkspace_js_1.canMoveToWorkspacePolicy)(loaders),
        canCreatePersonal: (0, canCreatePersonal_js_1.canCreatePersonalProjectPolicy)(loaders),
        canUpdate: (0, canUpdate_js_1.canUpdateProjectPolicy)(loaders),
        canDelete: (0, canDelete_js_2.canDeleteProjectPolicy)(loaders),
        canUpdateAllowPublicComments: (0, canUpdateAllowPublicComments_js_1.canUpdateProjectAllowPublicCommentsPolicy)(loaders),
        canReadSettings: (0, canReadSettings_js_1.canReadProjectSettingsPolicy)(loaders),
        canReadWebhooks: (0, canReadWebhooks_js_1.canReadProjectWebhooksPolicy)(loaders),
        canLeave: (0, canLeave_js_1.canLeaveProjectPolicy)(loaders),
        canInvite: (0, canInvite_js_2.canInviteToProjectPolicy)(loaders),
        canPublish: (0, canPublish_js_1.canPublishPolicy)(loaders),
        canLoad: (0, canLoad_js_1.canLoadPolicy)(loaders)
    },
    workspace: {
        canCreateProject: (0, canCreateWorkspaceProject_js_1.canCreateWorkspaceProjectPolicy)(loaders),
        canInvite: (0, canInvite_js_1.canInviteToWorkspacePolicy)(loaders),
        canReceiveProjectsUpdatedMessage: (0, canReceiveProjectsUpdatedMessage_js_1.canReceiveWorkspaceProjectsUpdatedMessagePolicy)(loaders),
        canUpdateEmbedOptions: (0, canUpdateEmbedOptions_js_1.canUpdateEmbedOptionsPolicy)(loaders)
    }
});
exports.authPoliciesFactory = authPoliciesFactory;
