// @speckle/shared/dist/esm/authz/domain/loaders.js
/**
 * All loaders must be listed here for app startup validation to work properly
 */
/* v8 ignore start  */
export const AuthCheckContextLoaderKeys = {
    getEnv: 'getEnv',
    getProject: 'getProject',
    getProjectRoleCounts: 'getProjectRoleCounts',
    getProjectRole: 'getProjectRole',
    getProjectModelCount: 'getProjectModelCount',
    getServerRole: 'getServerRole',
    getWorkspace: 'getWorkspace',
    getWorkspaceRole: 'getWorkspaceRole',
    getWorkspaceSeat: 'getWorkspaceSeat',
    getWorkspaceModelCount: 'getWorkspaceModelCount',
    getWorkspaceProjectCount: 'getWorkspaceProjectCount',
    getWorkspacePlan: 'getWorkspacePlan',
    getWorkspaceLimits: 'getWorkspaceLimits',
    getWorkspaceSsoProvider: 'getWorkspaceSsoProvider',
    getWorkspaceSsoSession: 'getWorkspaceSsoSession',
    getAdminOverrideEnabled: 'getAdminOverrideEnabled',
    getComment: 'getComment',
    getModel: 'getModel',
    getVersion: 'getVersion'
};
export const Loaders = AuthCheckContextLoaderKeys; // shorter alias
